lib/src/migrators/module.dart

```diff
@@ -782,22 +808,58 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
       configuration =
           " with (\n$indent  " + configured.join(',\n$indent  ') + "\n$indent)";
     }
-    if (!_useAllowed) {
+    if (!_state.useAllowed) {
       var namespace = _findOrAddBuiltInNamespace('meta');
       configuration = configuration.replaceFirst(' with', r', $with:');
       addPatch(Patch.insert(importStart,
           '@include $namespace.load-css(${import.span.text}$configuration)'));
     } else {
-      _usedUrls.add(resolvedUrl);
-      addPatch(Patch.insert(
-          importStart, '@use ${import.span.text}$asClause$configuration'));
+      _state.usedUrls.add(resolvedUrl);
+      var patch = Patch.insert(
+          importStart, '@use ${import.span.text}$asClause$configuration');
+      addPatch(patch);
+      if (_upstreamStylesheets.isEmpty &&
+          configuration.isEmpty &&
+          !_anyMemberReferenced(resolvedUrl)) {
+        _possibleForwardConversions[resolvedUrl] = patch;
+      }
     }
   }
 
+  /// Returns true if any member of [url] is referenced within the current
+  /// stylesheet.
+  bool _anyMemberReferenced(Uri url) {
+    for (var entry in references.variables.entries) {
+      if (entry.key.span.sourceUrl != currentUrl) continue;
+      var declaration = _reassignedVariables.contains(entry.value)
+          ? references.variableReassignments[entry.value]
+          : entry.value;
+      if (declaration.sourceUrl == url) return true;
+    }
+    for (var entry in references.variableReassignments.entries) {
+      if (entry.key.sourceUrl == currentUrl && entry.value.sourceUrl == url) {
+        return true;
+      }
+    }
+    for (var entry in references.mixins.entries) {
+      if (entry.key.span.sourceUrl == currentUrl &&
+          entry.value.sourceUrl == url) {
+        return true;
+      }
+    }
+    for (var entry in references.functions.entries) {
+      if (entry.key.span.sourceUrl == currentUrl &&
+          entry.value.sourceUrl == url) {
+        return true;
+      }
+    }
```

Comment:
Could you simplify this by defining `Set<SassNode> allReferences(MemberDeclaration declaration)` in `References`, and then iterating over `references.allDeclarations` rather than repeating a bunch of logic for each reference type? That helper would probably also let you simplify `References.referencedOutsideDeclaringStylesheet()`, and potentially other logic that wants to do the same thing for all members of all types.

Reply:
Done. I also changed this to be a method of `References`.